### ISSUE TITLE
Is the validation loss computed in the mmsegmentation 1.2.2? If so, how can i plot it?

### ISSUE THREAD

**Issue by HUANGL1NJIE** [open-mmlab/mmsegmentation#3494]
Hello,
I am using mmseg version 1.2.2.

I sincerely wonder how to get the validation loss and the accuracy score in the training step.

I found the `val_step` function was not in the` base.py` in this version. I am not sure whether the computation of validation loss is 

moved to other modules.

Thanks for any help.

**Comment by AI-Tianlong** [comment]
https://github.com/open-mmlab/mmsegmentation/blob/main/docs/en/advanced_guides/evaluation.md#ioumetric

**Comment by CastleDream** [comment]
看这个文档
<https://github.com/open-mmlab/mmsegmentation/blob/c685fe6767c4cadf6b051983ca6208f1b9d1ccb8/docs/zh_cn/advanced_guides/data_flow.md>

上图所示数据流仅适用于当用户没有自定义 Runner 中的 TrainLoop、ValLoop 和 TestLoop，并且没有在其自定义模型中覆写 train_step、val_step 和 test_step 方法时。MMSegmentation 中 loop 的默认设置如下：使用IterBasedTrainLoop 训练模型，共计 20000 次迭代，并且在每 2000 次迭代后进行一次验证
......

蓝色线表示 [val_step](https://github.com/open-mmlab/mmsegmentation/blob/c685fe6767c4cadf6b051983ca6208f1b9d1ccb8/docs/zh_cn/advanced_guides/models.md#val_step) 和 [test_step](https://github.com/open-mmlab/mmsegmentation/blob/c685fe6767c4cadf6b051983ca6208f1b9d1ccb8/docs/zh_cn/advanced_guides/models.md#test_step)。这两个过程的数据流除了模型输出与 train_step 不同外，其余均和 train_step 类似。由于在评估时模型参数会被冻结，因此模型的输出将被传递给 [Evaluator](https://github.com/open-mmlab/mmsegmentation/blob/c685fe6767c4cadf6b051983ca6208f1b9d1ccb8/docs/zh_cn/advanced_guides/evaluation.md#ioumetric)。 来计算指标。

也就是说，val的时候不会输出loss，会直接调用Evaluator计算iou等指标

**Comment by eighteenlin** [comment]
Hello,
I am also using mmseg version 1.2.2.
I sincerely wonder how to get the validation loss and the accuracy score in the training step.
I also encountered this problem, how did you solve it?
Thanks for any help.

**Comment by strangers66** [comment]
你好，这个问题解决了吗，我也想知道如何获得val loss

**Comment by CastleDream** [comment]
you can look this flow image in  [document](https://github.com/open-mmlab/mmsegmentation/blob/main/docs/zh_cn/advanced_guides/data_flow.md):

![](https://user-images.githubusercontent.com/112053249/199228350-5f80699e-7fd2-4b4c-ac32-0b16b1922c2e.png)

and when you in test and val stage, you just evaluator model, don't receive loss, and can't get val loss

if you want to get val loss, maybe you can see this issue: [Logging validation loss without library code hacks](https://github.com/open-mmlab/mmsegmentation/issues/1396#top)
#1396

**Comment by GZHU-SHM** [comment]
> https://github.com/open-mmlab/mmsegmentation/blob/main/docs/en/advanced_guides/evaluation.md#ioumetric Is this useful to you?

Sorry for the late reply. The tutorial from the link actually is more related to the evalution metric calculation. Maybe this comment https://github.com/open-mmlab/mmsegmentation/issues/3494#issuecomment-2136444244  understands my request.  Still, sincerely thanks for your comment.

**Comment by GZHU-SHM** [comment]
> 你好，这个问题解决了吗，我也想知道如何获得val loss

还没有，感觉会很麻烦。

**Comment by GZHU-SHM** [comment]
> Hello, I am also using mmseg version 1.2.2. I sincerely wonder how to get the validation loss and the accuracy score in the training step. I also encountered this problem, how did you solve it? Thanks for any help.

I am still searching for a effective solution, regret for not providing any useful help to you.

**Comment by GZHU-SHM** [comment]
> you can look this flow image in [document](https://github.com/open-mmlab/mmsegmentation/blob/main/docs/zh_cn/advanced_guides/data_flow.md):
> 
> ![](https://user-images.githubusercontent.com/112053249/199228350-5f80699e-7fd2-4b4c-ac32-0b16b1922c2e.png)
> 
> and when you in test and val stage, you just evaluator model, don't receive loss, and can't get val loss
> 
> if you want to get val loss, maybe you can see this issue: [Logging validation loss without library code hacks](https://github.com/open-mmlab/mmsegmentation/issues/1396#top) #1396

Thanks for your patient comments. From the flow image, it seems like the val loss is unable to be calculated in the mmseg 1.2.2. The solutions discussed in the issue may only work with earlier versions of mmseg.

**Comment by syf66666666** [comment]
> > 您可以在[文档](https://github.com/open-mmlab/mmsegmentation/blob/main/docs/zh_cn/advanced_guides/data_flow.md)中查看此流程图像：
> > ![](https://user-images.githubusercontent.com/112053249/199228350-5f80699e-7fd2-4b4c-ac32-0b16b1922c2e.png)
> > 而当你处于测试和Val阶段时，你只是评估者模型，没有收到损失，也无法获得Val损失
> > 如果你想获得 val 损失，也许你可以看到这个问题： [在没有库代码黑客的情况下记录验证丢失](https://github.com/open-mmlab/mmsegmentation/issues/1396#top) #1396
> 
> 感谢您的耐心评论。从流量图像来看，似乎无法在 mmseg 1.2.2 中计算 val 损失。本问题中讨论的解决方案可能仅适用于早期版本的 mmseg。

我0.30.0版本的使用这个方法有用
